### PR TITLE
Remove akka from runtime

### LINKED ISSUE #8222: Remove akka dependencies from `runtime` project
`runtime` project, and thus `runtime.jar` fat jar, transitively contains akka dependencies, for example from `runtime` --> `downloader` --> `akka`. We should remove akka from `downloader` and implement the downloading by some simple http component from JDK (or Scala's standard library).

`akka` is unnecessarily heavy-weight, it comprises of a lot of classes that are loaded just because of few simple HTTP requests.
```[tasklist]
### Tasks
- [x] Ensure there is are no `akka` classes in `runtime.jar`
- [x] Remove `akka` from `downloader` (#8953)
- [x] Split `connected-lock-manager` into two projects
```

### PR DESCRIPTION
**Pull request by Akirathan**
Closes #8222 

### Pull Request Description

There are two projects transitively required by `runtime`, that have akka dependencies:
- `downloader`
- `connected-lock-manager`

This PR replaces the `akka-http` dependency in `downloader` by HttpClient from JDK, and splits `connected-lock-manager` into two projects such that there are no akka classes in `runtime.jar`.

### Important Notes

- Simplify the `downloader` project - remove akka.
- Add HTTP tests to the `downloader` project that uses our `http-test-helper` that is normally used for stdlib tests.
 - It required few tweaks so that we can embed that server in a unit test.
- Split `connected-lock-manager` project into two projects - remove akka from `runtime`. 
- **Native image build fixes and quality of life improvements:**
  - Output of `native-image` is captured 743e167aa4e795d5fdb3feaab124de8994202189
    - The output will no longer be intertwined with the output from other commands on the CI.
  - Arguments to the `native-image` are passed via an argument file, not via command line - ba0a69de6e715340dfc912368054dfbc9c48ad57
    - This resolves an issue on Windows with "Command line too long", for example in https://github.com/enso-org/enso/actions/runs/7934447148/job/21665456738?pr=8953#step:8:2269

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

**Comment by hubertp** [comment]
Are you aware of https://github.com/enso-org/enso/issues/8934? You can probably more or less revert the commit that introduced it.

**Comment by Akirathan** [comment]
> Are you aware of #8934? You can probably more or less revert the commit that introduced it.

Thanks for pointing that out. Let's first try to use only JDK. I think that the HttpClient provided by the JDK is more than capable of handling few requests. There is no need to add yet another dependency.

File: engine/runtime/src/test/java/org/enso/runtime/test/DependenciesTest.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import static org.junit.Assert.fail;
+
+import org.junit.Test;
+
+public class DependenciesTest {
+  @Test
+  public void noAkkaDependencies() {
+    try {
+      Class.forName("akka.actor.ActorSystem$");
+      fail("akka.actor.ActorSystem should not be on the classpath");
```
Comment (Akirathan) [inline engine/runtime/src/test/java/org/enso/runtime/test/DependenciesTest.java:12]: This test is a bit clumsy. If `akka.actor.ActorSystem` is not on the class-path, it only means that `runtime/test` does not have `akka` dependency, it does not check that there are no akka classes in `runtime.jar`. However, I still think that this test is useful. Moreover, I have manually checked that `runtime.jar` does not contain any akka classes.

File: lib/scala/downloader/src/main/scala/org/enso/downloader/http/HTTPRequest.scala [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package org.enso.downloader.http
 
-import akka.http.scaladsl.model.HttpRequest
+import java.net.http.HttpRequest
```
Comment (JaroslavTulach) [inline lib/scala/downloader/src/main/scala/org/enso/downloader/http/HTTPRequest.scala:3]: Using class from JDK is a great step forward!

**Review by JaroslavTulach — APPROVED** [review]

File: lib/scala/downloader/src/main/java/org/enso/downloader/http/FilesMultipartBodyPublisher.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.List;
+
+/**
+ * This utility class adds support for {@code multipart/form-data} content-type as there is no such
+ * support in {@code java.net.http} JDK.
+ *
+ * <p>Inspired by <a href="https://stackoverflow.com/a/56482187/4816269">SO</a>.
+ */
+public class FilesMultipartBodyPublisher {
+  public static HttpRequest.BodyPublisher ofMimeMultipartData(
```
Comment (radeusgd) [inline lib/scala/downloader/src/main/java/org/enso/downloader/http/FilesMultipartBodyPublisher.java:19]: I don't like us implementing such utilities ourselves instead of using existing well-tested solutions.

This works nicely for simple cases, but how do we guarantee that it will work well in general? Should we try testing it as well as an existing solution would? That adds a lot of maintenance burden on us.

> Example 1: the boundary.

I see it is generated as a UUID. I guess it is extremely unlikely that the same UUID would appear inside of the contents of one of the parts submitted as part of this form, but it is not 100% guaranteed.

I'm a bit nitpicky here as indeed it is going to be **very** unlikely to run into this kind of collision. But was this even considered when this code was written?

This is a seemingly simple operation but it may raise many uncertainities like the example above. IMO relying on an existing solution is better as it does not put the maintenance burden of ensuring this solution is bug free, on us.

File: lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala [hunk 4]
```diff
@@ -1,33 +1,42 @@
...
+  def addQuery(key: String, value: String): URIBuilder = {
+    val scheme    = uri.getScheme
+    val authority = uri.getAuthority
+    val path      = uri.getPath
+    val query     = if (uri.getQuery == null) "" else uri.getQuery + "&"
+    val fragment  = uri.getFragment
+    val newQuery  = query + key + "=" + value
+    val newUri    = new URI(scheme, authority, path, newQuery, fragment)
+    copy(newUri)
+  }
```
Comment (radeusgd) [inline lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala]: What about encoding the `key` and `value`?

Again, replacing an existing solution with a self-built one puts it on us to ensure that it works also in irregular cases.

Well, I don't want to be complaining too much - with where this downloader is currently being used I guess maybe a non-encoding query builder will work OK. But the moment a non-expected character shows up e.g. in a version number or username, it may break. That's why I think using existing solutions is better.

----

Of course not suggesting to stay with Akka, it is an overkill for sure.

But maybe we should use Apache HttpCore for these builders?
Or at least let's either do proper encoding or admit we can't handle non-standard characters and add a check that the key/value only contains `[A-Za-z0-9-_]` or something like that. If stuff breaks, at least it will be clearer why.

**Review by radeusgd — APPROVED** [review]
Looks good.

I'm a bit worried about rolling our own URI and Multipart body handling - these seem like they are simple, but there are traps hidden there.

But I don't want to be complaining too much - we should be fine with these for now and they are not heavily used anyway.

I'd appreciate if we put a comment for future maybe, so that once we start using these more, we could revisit and see if any 'reinforcements' are needed there.

File: lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala (hunk 4 above)
Comment (radeusgd) [reply lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala]: Btw. we do have our own classes for handling these.

I actually did break what I say here that 'I prefer to use existing solutions instead of maintaining our own' because there was pressure to try avoiding the Apache Http as dependency of `Standard.Base`.

But we do [have tests for lots of various edge cases there](https://github.com/enso-org/enso/blob/develop/test/Base_Tests/src/Network/URI_Spec.enso#L135-L149).

---

Maybe you could refactor the common parts (perhaps `URITransformer`) into a separate library that can be used from both places?

**Comment by hubertp** [comment]
@radeusgd Didn't you mention that at some point you moved from JDK to Apache because of some bug in the former? Is that fixed now?

**Comment by radeusgd** [comment]
> @radeusgd Didn't you mention that at some point you moved from JDK to Apache because of some bug in the former? Is that fixed now?

IIRC that was this bug https://bugs.openjdk.org/browse/JDK-8231449 it seems to indeed be fixed in Java 14 (we've been using Java 11 back then and I think it was never backported to versions <13).

File: lib/scala/downloader/src/main/java/org/enso/downloader/http/FilesMultipartBodyPublisher.java (hunk 3 above)
Comment (Akirathan) [reply lib/scala/downloader/src/main/java/org/enso/downloader/http/FilesMultipartBodyPublisher.java:19]: The `downloader` project makes only a very specific kind of HTTP requests and expects a very specific and narrow set of HTTP responses. We are in a full control over these reqs and responses. This project is only used for library uploading and downloading, which is also covered by tests.

Implementing a very simple handling of `multipart/form-data` is a minimal price we have to pay here. Please keep in mind that we are trying very hard, in general, to reduce the number of classes that will eventually get into `runtime.jar`.

File: lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala (hunk 4 above)
Comment (Akirathan) [reply lib/scala/downloader/src/main/scala/org/enso/downloader/http/URIBuilder.scala]: Fixed in c5cc2dc4481ff64ab6c0b8bb4fb8da55001eab58
